### PR TITLE
add support for rspec mirroring existing cucumber implementation

### PR DESCRIPTION
**Pull request by Castone22**
Adds basic support for RSpec that behaves in a way that should be familiar to the existing cucumber implementation.

Also adds some examples of its use to the example project.

Some changes were made to cucumber to make this work.  One of the big ones is that if rspec is kept in the same project directory as cucumber you have to exclude the spec directory or cucumber tries to run the specs.

I also refactored how the app name is handled in the setup, and moved the setup into the support directory and prefixed it with an underscore so that cucumber would load it automatically before it loads even the env.rb file.  The .rspec file still have to manually require the setup file however.   This makes it a bit easier to run cucumber tests using rubymine.

Due to some idiosyncrasies with how rspec determines a scenario passed, i can't get this information in a hook (rspec treats the hooks as part of the scenario and will mark a scenario as failing if any of them fail, therefore scenario status is not set until after the world is discarded.  More on that later.) and had to define a formatter.  Because rspec destroys the example's world before calling the formatter, I can't retrieve the logger from the world anymore so i have to attach it to the example class itself and pull it out in the formatter using instance variable set and get.  It's not pretty and I certainly need to deliberate on a better way to accomplish this, but it was the best way i could think of to replicate the existing logging behavior. 

This was mostly done because I feel that RSpec falls more in line with the design ideologies that are being communicated for Oz than cucumber.

### CODE REVIEW

File: CORE/spec/formatters/status_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  RSpec::Core::Formatters.register self, :example_passed, :example_pending, :example_failed
+
+  def initialize(*args)
+  end
+
+  def example_finished(notification)
+    example = notification.example
+    # For Some reason rspec blows away the instance before this point, so i have to persist the logger in the example which isn't really
+    # a defined rspec behavior...
+    @logger = example.instance_variable_get(:@logger)
```
Comment (Castone22) [inline CORE/spec/formatters/status_formatter.rb]: flagging this line as mentioned in the PR body

File: CORE/spec/hooks/hooks.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    create_world
+    log_spec(example)
+  end
+
+  c.after(:each) do |example|
+    cleanup_world
+    # For Some reason rspec blows away the instance before the formatter is called, so i have to persist the logger in the example which isn't really
+    # a defined rspec behavior...
+    example.instance_variable_set(:@logger, example.example_group_instance.logger)
+  end
```
Comment (Castone22) [inline CORE/spec/hooks/hooks.rb]: Also flagging this line as mentioned in the PR body.

File: CORE/elements/_core_element.rb [hunk 3]
```diff
@@ -104,7 +104,7 @@ def visible?
 
   def present?
     begin
-      watir_element.wait_until_present(timeout:0)
+      watir_element.wait_until(timeout: 0, &:present?)
```
Comment (dbwest) [inline CORE/elements/_core_element.rb]: why this change?

Comment (Castone22) [reply CORE/elements/_core_element.rb]: Updating to the latest watir standard to remove a warning

File: CORE/spec/helpers/page_helper.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+module Oz
+  module PageHelper
+    def start_at(target_page, from_page: nil)
+      @root_page.begin_new_session
+      navigate_to(target_page, from_page: from_page)
+    end
+
+    def navigate_to(target_page, from_page: nil)
+      from_page = CoreUtils.find_class(from_page&.gsub(' ', '::') + ' Page') if from_page&.is_a?(Class)
+      target_page = CoreUtils.find_class(target_page.gsub(' ', '::') + ' Page') unless target_page.is_a? Class
```
Comment (dbwest) [inline CORE/spec/helpers/page_helper.rb]: Why lines 9 and 10?

File: CORE/spec/helpers/page_helper.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+module Oz
+  module PageHelper
+    def start_at(target_page, from_page: nil)
```
Comment (dbwest) [inline CORE/spec/helpers/page_helper.rb]: should this be `from_page = nil`
